src/bit_machine/exec.rs

```diff
@@ -410,8 +410,9 @@ impl BitMachine {
                 }
                 RedeemNodeInner::Witness(value) => self.write_value(value),
                 RedeemNodeInner::Hidden(..) => return Err(ExecutionError::ReachedPrunedBranch),
-                RedeemNodeInner::Jet(j) => App::exec_jet(j, self, env)
-                    .map_err(|x| ExecutionError::AppError(Box::new(x)))?,
+                RedeemNodeInner::Jet(j) => {
+                    App::exec_jet(j, self, env).map_err(ExecutionError::from)?
```

Comment:
In 6ec8e71cee1906a73fb3dc5c2601e15cd6dc256c:

This call to `from` is unnecessary. `?` will call it for you.